### PR TITLE
[Shaman] Update Elemental ST APL for TWW s1.

### PR DESCRIPTION
**Pull request by MadBubble**
1) Removed bag of tricks: dps loss
2) Removed Ascendance condition from PI: Ascendance isnt big enough to delay
3) Cut all "how many targets together" conditions to just "CL hits > X": if CL hits X then active enemies are at least X anyway, LvBm check is redundant, sim can check CL version during Ascendance
4) Limit pwave to not waste surge if FS buff is rolling in st 
5) Simplify Stormkeeper to just "dont cast during Ascendance" and "dont overwrite existing SK buff" 
6) Echo chamber is enough to use LB with SoP outside of Ascendance even in fire builds, not Lightning exclusive anymore 
7) Moved Ascendance below all that SK flock
8) Added new block of Mote'd spenders acting as a primary way to spend 
9) Unified all other spender lines in a block in weaving section 
10) EQ is worth with EoGS even in st with EB talented so cut few conditions here and there 
11) Added new Icefury logic: Dont waste stacks during Ascendance, use all stacks (in weaving manner if possible) as first prio after spending 
12) Cleaned the bottom, now it is just essentially a prio list for MotE buffs and getting rid of procs. For lightning "buff spender" line is acting like a cycle breaker

### CODE REVIEW

**Comment by MadBubble** [comment]
@wfarr @HawkCorrigan @Bloodmallet

File: engine/class_modules/sc_shaman.cpp
Comment (ikantarelis) [inline engine/class_modules/sc_shaman.cpp]:
```suggestion
    aoe->add_action( "totemic_recall,if=cooldown.liquid_magma_totem.remains>25", "Reset LMT CD as early as possible." );
```

File: engine/class_modules/sc_shaman.cpp
Comment (wfarr) [inline engine/class_modules/sc_shaman.cpp]: You've got hard tabstops here instead of spaces. Ditto in a few other spots.

File: engine/class_modules/sc_shaman.cpp
Comment (wfarr) [inline engine/class_modules/sc_shaman.cpp:12952]: We can optimize this a bit later, but I wouldn't be surprised if there's same gain to be had ignoring FlS refresh for 1-2 GCDs if PWave or LMT are coming up soon.

File: engine/class_modules/sc_shaman.cpp
Comment (wfarr) [inline engine/class_modules/sc_shaman.cpp:12961]: Confirming: SK CL on 2 > SK LB?

**Review by wfarr — APPROVED** [review]
A few questions that can be chased down later and don't need to block merging this.

A few whitespace changes it would be good to sort out before merging to eliminate noisy diffs in subsequent changes (mostly replacing hard tabs with soft tabs type issues).

File: engine/class_modules/sc_shaman.cpp
Comment (MadBubble) [reply engine/class_modules/sc_shaman.cpp:12961]: anything related to 2t or aoe is not part of this PR, i just left whatever was in place for those scenarios coz we only got proper Lava surge logic in now. It will be separate update.

File: engine/class_modules/sc_shaman.cpp
Comment (MadBubble) [reply engine/class_modules/sc_shaman.cpp:12952]: rn it is just backup line for fire to not get lost if smth goes wrong in low gear or no BL sim. It is restricted to fire only and with proper gear never fires up coz fire elem uptime is basically 100% and pwave cd is shorter than fs duration

**Pull request merged**
